src/meta/src/hummock/hummock_manager.rs

```diff
@@ -940,6 +943,21 @@ where
                 )?;
             }
             versioning.current_version = new_version;
+
+            let worker_nodes = self
+                .cluster_manager
+                .list_worker_node(WorkerType::ComputeNode, None)
+                .await;
+            for worker_node in worker_nodes {
+                if let Some(min_pinned_id) = versioning
+                    .pinned_versions
+                    .get(&worker_node.id)
+                    .map(|pinned_version| pinned_version.min_pinned_id)
+                {
+                    host2pin.push((WorkerKey(worker_node.host.unwrap()), min_pinned_id));
+                }
+            }
+            deltas_to_send = versioning.hummock_version_deltas.clone();
```

Comment:
Can we calculate the real deltas_to_send right here, rather than sending host2pin and calculating it later in notify?

Reply:
> Can we calculate the real deltas_to_send right here, rather than sending host2pin and calculating it later in notify?

unfortunately we need to send different CNs different range of `hummock_version_delta`s

Reply:
Does it mean that we must send all history `hummock_version_deltas` to every node every time we update a new version?

Reply:
> Does it mean that we must send all history `hummock_version_deltas` to every node every time we update a new version?

range(`min_pinned_version` of that node, current_version_id)

Reply:
Could we clone `min(min_pinned_id)` for `deltas_to_send`? 
Because `hummock_version_deltas` may be large if we does not GC it in time.

Reply:
I means we only need to clone range(min(min_pinned_id)..) for `deltas_to_send`. And if some worker_node does not own `pinned_version` or `min_pinned_id` of it is smaller than the minimal delta id in `versioning.hummock_version_deltas`,  We shall notify it with a whole `HummockVersion`.

Reply:
> I means we only need to clone range(min(min_pinned_id)..) for `deltas_to_send`. And if some worker_node does not own `pinned_version` or `min_pinned_id` of it is smaller than the minimal delta id in `versioning.hummock_version_deltas`, We shall notify it with a whole `HummockVersion`.

this scheme will send more deltas than this PR does

Reply:
No. If we clone all the deltas every times report compact task, it will cost more resouce and increase latency for this operation.  I think we shall run a benchmark (more than 10 minutes) to see whether this PR would cause performance regression.

Reply:
Because of write-amplification, the size of deltas could be much larger than the origin `HummockVersion`.

Reply:
> Could we clone `min(min_pinned_id)` for `deltas_to_send`?
> Because `hummock_version_deltas` may be large if we does not GC it in time.

just fixed code, now it calculates `min(min_pinned_id)`